toolchain/parse/state.def

```diff
@@ -707,7 +707,7 @@ CARBON_PARSE_STATE_VARIANTS2(ParenConditionFinish, If, While)
 // ^
 //   1. Expr
 //   2. ParenExprParamFinishAsUnknown
-//   3. ParenExprFinishAsNormal    (SPECIAL: may be replaced)
+//   3. ParenExprFinishAsGrouping    (SPECIAL: may be replaced)
```

Comment:
> Where does the term "grouping" for this come from? I'm not familiar with this usage of it, and checking [expressions](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/expressions/README.md) I'm not seeing mentions of it.

See e.g. [here](https://en.wikipedia.org/wiki/Symbols_of_grouping), especially "For example, in the expression 3(x+y) the parentheses are symbols of grouping, but in the expression (3, 5) the parentheses may indicate an open interval."

> Similarly in the parse tree, we would refer to this case as "ParenExpr" (as in "ParenExprOrTupleLiteralStart"; "ParenExprAsParenExpr" probably felt redundant, but "ParensAs*" would've been an alternative [though it loses useful Expr context]). "Grouping" this would make me expect "GroupingOrTupleLiteralStart".

That's a good point. I suggest using "paren expression" to refer to any expression delimited by parentheses, and "grouping expression" to refer to a paren expression that isn't a tuple literal. I've tried to apply that change consistently throughout.

Reply:
How about separating this out, or getting a lead to review? I don't want the implementation getting out of sync with Carbon's own documentation.

Reply:
And to be sure, here are a few examples of "parenthesized expression": https://github.com/search?q=repo%3Acarbon-language%2Fcarbon-lang+%22parenthesized+expression%22&type=code

The similar search for "grouping expression" shows no results, so I think that's a naming change, not a cleanup. https://github.com/search?q=repo%3Acarbon-language%2Fcarbon-lang+%22grouping+expression%22&type=code

Reply:
OK, I've gone in the other direction, consistently using `ParenExpr` only when it doesn't refer to a tuple literal, and using `ParenExprOrTupleLiteral` as the inclusive term. I'll pursue changing that terminology separately.

Reply:
Just to be sure: while I'm fine with the PatternBinding rename, I'd like to avoid churn related to ParenExpr. Right now I'm going to wait for a resolution to #3417, but I'd be happy too if you wanted to change this to only be the PatternBinding rename. That'd mean all the misc test file changes wouldn't be combined with unrelated changes, also making it a little easier to review.

Reply:
Sure, I'm happy to wait for #3417

Reply:
I've now updated this based on the resolution of #3417.